lib/platform/gitea/index.ts

```diff
@@ -0,0 +1,361 @@
+import URL from 'url';
+import is from '@sindresorhus/is';
+
+import api from './gl-got-wrapper';
+import * as hostRules from '../../util/host-rules';
+import GitStorage from '../git/storage';
+import { PlatformConfig } from '../common';
+
+let config: {
+  storage: GitStorage;
+  repository: string;
+  localDir: string;
+  defaultBranch: string;
+  baseBranch: string;
+  email: string;
+  prList: any[];
+  issueList: any[];
+  token: string;
+} = {} as any;
+
+const defaults = {
+  hostType: 'gitea',
+  endpoint: 'https://gitea.com/api/v1/',
+};
+
+export async function initPlatform({
+  endpoint,
+  token,
+}: {
+  endpoint: string;
+  token: string;
+}) {
+  if (!token) {
+    throw new Error('Init: You must configure a Gitea personal access token');
+  }
+  const res = {} as any;
+  if (endpoint) {
+    res.endpoint = endpoint.replace(/\/?$/, '/'); // always add a trailing slash
+    api.setBaseUrl(res.endpoint);
+    defaults.endpoint = res.endpoint;
+  } else {
+    res.endpoint = defaults.endpoint;
+    logger.info('Using default Gitea endpoint: ' + res.endpoint);
+  }
+  try {
+    res.gitAuthor = (await api.get(`user?token=${token}`)).body.email;
+  } catch (err) {
+    logger.info({ err }, 'Error authenticating with Gitea. Check your token');
+    throw new Error('Init: Authentication failure');
+  }
+  return res;
+}
+
+// Get all repositories that the user has access to
+export async function getRepos() {
+  logger.info('Autodiscovering Gitea repositories');
+  try {
+    // TODO : check paginate capabilities
+    // TODO : limit to user repositories (add uid parameter ?)
+    // const url = `repos?membership=true&per_page=100`;
+    const url = `repos/search/`;
```

Comment:
Does this return only repos that the authenticated user has access to? Or possibly all repos on the system?